nixos/modules/services/networking/aria2.nix

```diff
@@ -117,6 +120,7 @@ in
           touch "${sessionFile}"
         fi
         cp -f "${settingsFile}" "${settingsDir}/aria2.conf"
+        echo "rpc-secret=$(cat ${escapeShellArg cfg.rpcSecretFile})" >> "${settingsDir}/aria2.conf"
```

Comment:
Could you pass the secret via systemd `LoadCredential`? The `aria2` user might not have the ability to read it.